tools/deb/test/run_gce_test.sh

```diff
@@ -0,0 +1,351 @@
+function istioVMExternalIP() {
+  local NAME=${1:-$TESTVM}
+  gcloud compute --project $PROJECT instances describe $NAME --zone $ISTIO_ZONE --format='value(networkInterfaces[0].accessConfigs[0].natIP)'
+}
+
+function istioVMInternalIP() {
+  local NAME=${1:-$TESTVM}
+  gcloud compute --project $PROJECT instances describe $NAME  --zone $ISTIO_ZONE --format='value(networkInterfaces[0].networkIP)'
+}
+
+# Initialize the K8S cluster, generating config files for the raw VMs.
+# Must be run once, will generate files in the CWD. The files must be installed on the VM.
+# This assumes the recommended dnsmasq config option.
+function istioPrepareCluster() {
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: istio-pilot-ilb
+  annotations:
+    cloud.google.com/load-balancer-type: "internal"
+  labels:
+    istio: pilot
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 8080
+    protocol: TCP
+  selector:
+    istio: pilot
+EOF
+cat <<EOF | kubectl apply -n kube-system -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: dns-ilb
+  annotations:
+    cloud.google.com/load-balancer-type: "internal"
+  labels:
+    k8s-app: kube-dns
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 53
+    protocol: UDP
+  selector:
+    k8s-app: kube-dns
+EOF
+
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: mixer-ilb
+  annotations:
+    cloud.google.com/load-balancer-type: "internal"
+  labels:
+    istio: mixer
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 8000
+    protocol: TCP
+  selector:
+    istio: mixer
+EOF
+
+  for i in {1..10}
+  do
+    PILOT_IP=$(kubectl get service istio-pilot-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    ISTIO_DNS=$(kubectl get -n kube-system service dns-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    MIXER_IP=$(kubectl get service mixer-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    if [ ${PILOT_IP} == "" -o  ${PILOT_IP} == "" -o ${MIXER_IP} == "" ] ; then
+        echo Waiting for ILBs
+        sleep 10
+    else
+        break
+    fi
+  done
+
+  if [ ${PILOT_IP} == "" -o  ${PILOT_IP} == "" -o ${MIXER_IP} == "" ] ; then
+    echo "Failed to create ILBs"
+    exit 1
+  fi
+
+  #/etc/dnsmasq.d/kubedns
+  echo "server=/default.svc.cluster.local/$ISTIO_DNS" > kubedns
+  echo "address=/istio-mixer/$MIXER_IP" >> kubedns
+  echo "address=/istio-pilot/$PILOT_IP" >> kubedns
+
+  CIDR=$(gcloud container clusters describe ${K8SCLUSTER} --zone=${ISTIO_ZONE} --format "value(servicesIpv4Cidr)")
+  echo "ISTIO_SERVICE_CIDR=$CIDR" > cluster.env
+
+}
+
+# Configure a service running on the VM.
+# Params:
+# - port of the service
+# - service name (default to raw vm name)
+# - IP of the rawvm (will get it from gcloud if not set)
+#
+function istioConfigHttpService() {
+  local NAME=${1:-}
+  local PORT=${2:-}
+  local IP=${3:-}
+
+  # The 'name: http' is critical - without it the service is exposed as TCP
+
+  cat << EOF | kubectl apply -f -
```

Comment:
to be replaced by istioctl register (https://github.com/istio/pilot/pull/1100) ideally (though that means building or downloading istioctl)

Reply:
Ack, we should add it in a separate PR (both kubeclt and istioctl need to be tested). 

Note that for istioctl we also want to test that it can be run from the VM itself, as we discussed for the DHCP/non fixed address. And in context of the new consul updates, we probably need to extend it to support consul as well ( in 0.3)

Reply:
if kubectl can be run, istioctl can be run

consul has it's own registration but yes in 0.3 we should look into unification, not in 0.2

ok to postpone using istioctl here for right now but any suggestion on how we'd get istioctl binary here ?